src/Shared/ui/CustomSelectComponent/CustomSelectComponent.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  FormHelperText,
+  InputLabel,
+  SelectChangeEvent,
+  useTheme,
+} from '@mui/material'
+import {CustomFormControl, CustomMenuItem, CustomSelect} from './StyledElement'
+import {useTranslation} from 'react-i18next'
+import {FC, forwardRef} from 'react'
+
+interface CustomSelectComponentProps {
+  value: string
+  onChange: (event: SelectChangeEvent<unknown>, child: React.ReactNode) => void
+  options: {value: string; label: string}[]
+  label: string
+  helpingText?: string
+  disabled?: boolean
+  defaultValue?: string
+}
```

Comment:
Вместо `unknown` должен быть generic type
```suggestion
interface CustomSelectComponentProps <T>{
  value: string
  onChange: (event: SelectChangeEvent<T>, child: React.ReactNode) => void
  options: {value: string; label: string}[]
  label: string
  helpingText?: string
  disabled?: boolean
  defaultValue?: string
}
```
Соответсвенно еще поменять остальные типы и убрать для них `as string` и тд